gdal/src/main/scala/geotrellis/contrib/vlm/gdal/GDALDataPath.scala

```diff
@@ -0,0 +1,30 @@
+package geotrellis.contrib.vlm.gdal
+
+import geotrellis.contrib.vlm.DataPath
+
+import java.net.MalformedURLException
+
+
+case class GDALDataPath(val path: String) extends DataPath {
+  protected def servicePrefix: String = "gdal+"
+
+  private lazy val strippedPath: String =
+    if (path.startsWith(servicePrefix))
+      path.splitAt(servicePrefix.size)._2
+    else
+      path
+
+  private lazy val vsiPath: VSIPath = VSIPath(strippedPath)
```

Comment:
Can `GDALDataPath` and `VSIPath` be the same thing? They're both basically "case class that is understood by GDAL layer and handles parsing and checking".